src/test/util/cluster_linearize.h

```diff
@@ -235,27 +253,63 @@ struct DepGraphFormatter
             assert(reordering.size() < SetType::Size());
             auto topo_idx = topo_depgraph.AddTransaction(new_feerate);
             topo_depgraph.AddDependencies(new_ancestors, topo_idx);
-            diff %= total_size + 1;
-            // Insert the new transaction at distance diff back from the end.
-            for (auto& pos : reordering) {
-                pos += (pos >= total_size - diff);
+            if (total_size < SetType::Size()) {
+                // Normal case.
+                diff %= SetType::Size();
+                if (diff <= total_size) {
+                    // Insert the new transaction at distance diff back from the end.
+                    for (auto& pos : reordering) {
+                        pos += (pos >= total_size - diff);
+                    }
+                    reordering.push_back(total_size++ - diff);
+                } else {
+                    // Append diff - total_size holes at the end, plus the new transaction.
+                    total_size = diff;
+                    reordering.push_back(total_size++);
+                }
+            } else {
+                // In case total_size == SetType::Size, it is not possible to insert the new
+                // transaction without exceeding SetType's size. Instead, interpret diff as an
+                // index into the holes, and overwrite a position there. This branch is never used
+                // when deserializing the output of the serializer, but gives meaning to otherwise
+                // invalid input.
+                diff %= (SetType::Size() - reordering.size());
+                SetType holes = SetType::Fill(SetType::Size());
+                for (auto pos : reordering) holes.Reset(pos);
+                for (auto pos : holes) {
+                    if (diff == 0) {
+                        reordering.push_back(pos);
```

Comment:
Haven't looked close enough to the new format, but pushing another element to `reordering` after hitting `SetType::Size` already seems to make the DepGraph construction later odd? Couldn't this make the mapping argument too large?

Reply:
Ah, missed the line where it was being set to `diff`.